src/experimental/utils.cpp

```diff
@@ -0,0 +1,242 @@
+             * Arguments:
+                * grid [Eigen::Matrix]: Grid for cleaning
+            * Return:
+                * Void
+             */
+            grid.setZero();
+        }
+
+
+        void clearVisitedCells(Eigen::MatrixXi & grid)
+        {
+            /**
+             * Clear the given integer Eigen::Matrix
+             * Arguments:
+                * grid [Eigen::Matrix]: Grid for cleaning
+             * Return:
+                * Void
+             */
+
+            grid.setZero();
+        }
+
+
+        karto::Vector2<int> getGridPosition(karto::Vector2<kt_double> const& position, kt_double resolution)
+        {
+            /**
+             * Mapping a continuous position into a grid position
+             * Arguments:
+                * pose [karto::Vector2<kt_double>]: Continuos pose
+                * resolution [kt_double]: Cell resolution
+             * Return:
+                * karto::Vector2<int>: Grid position
+             */
+            int x_cell = floor((position.GetX() / resolution));
+            int y_cell = floor((position.GetY() / resolution));
+
+            return karto::Vector2<int>{x_cell, y_cell};
+        }
+
+
+        karto::Vector2<kt_double> calculateCellIntersectionPoints(karto::Vector2<kt_double> const & laser_start,
+            karto::Vector2<kt_double> const & laser_end, karto::Vector2<kt_double> const & cell_start, karto::Vector2<kt_double> const & cell_end)
+        {
+            /**
+             * Find the intersection point between a cell line and a laser beam
+             * Arguments:
+                * laser_start [karto::Vector2<kt_double>]: Laser initial point in x and y
+                * laser_end [karto::Vector2<kt_double>]: Laser final point in x and y
+                * cell_start [karto::Vector2<kt_double>]: Cell initial point in x and y
+                * cell_end [karto::Vector2<kt_double>]: Cell final point in x and y
+             * Return:
+                * karto::Vector2<kt_double>: Intersection point
+             */
+
+            kt_double x1 = laser_start.GetX();
+            kt_double x2 = laser_end.GetX();
+            kt_double x3 = cell_start.GetX();
+            kt_double x4 = cell_end.GetX();
+
+            kt_double y1 = laser_start.GetY();
+            kt_double y2 = laser_end.GetY();
+            kt_double y3 = cell_start.GetY();
+            kt_double y4 = cell_end.GetY();
+
+            kt_double den = ((x2 - x1)*(y4 - y3) - (x4 - x3)*(y2 - y1));
+
+            karto::Vector2<kt_double> intersection;
+            if (den == 0.0f)
+            {
+                return{};
+            }
+            else
+            {
+                kt_double x = ((x2*y1 - x1*y2)*(x4 - x3) - (x4*y3 - x3*y4)*(x2 - x1)) / den;
+                kt_double y = ((x2*y1 - x1*y2)*(y4 - y3) - (x4*y3 - x3*y4)*(y2 - y1)) / den;
+                intersection.SetX(x);
+                intersection.SetY(y);
+            }
```

Comment:
@kmilo7204 nit: perhaps
```suggestion
            if (den != 0.0f)
            {
                kt_double x = ((x2*y1 - x1*y2)*(x4 - x3) - (x4*y3 - x3*y4)*(x2 - x1)) / den;
                kt_double y = ((x2*y1 - x1*y2)*(y4 - y3) - (x4*y3 - x3*y4)*(y2 - y1)) / den;
                intersection.SetX(x);
                intersection.SetY(y);
            }
```
reads better?